include/at_cmd_interface.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file at_cmd_interface.h
+ *
+ * @brief Public APIs for the AT command interface driver.
+ */
+
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#ifndef ZEPHYR_INCLUDE_AT_INTERFACE_H_
+#define ZEPHYR_INCLUDE_AT_INTERFACE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+
+/**
+ * @brief AT command return codes
+ */
+enum at_cmd_return_code {
+	AT_CMD_OK,
+	AT_CMD_ERROR,
+	AT_CMD_CMS,
+	AT_CMD_CME,
+	AT_CMD_NOT_RETURN_CODE,
+};
+
+/**
+ * @typedefs at_cmd_interface_handler_t
+ * @brief Callback API when a message is received from the mode
+ *
+ * @param "char * response" Null terminated string containing the modem message
+ * @param "size_t response_len" Length of string
+ *
+ */
+typedef void (*at_cmd_interface_handler_t)(char *response,
+					   size_t response_len);
+
+/**
+ * @brief Function to send an AT command to the modem
+ *
+ * @param cmd Pointer to null terminated AT command string
+ */
+enum at_cmd_return_code at_cmd_interface_write(const char *const cmd);
+
+/**
+ * @brief Function to set AT command message handler
+ *
+ * @param handler Pointer to a received messages handler function
+ */
+void                    at_cmd_interface_set_handler(at_cmd_interface_handler_t
+							handler);
+
+/**
+ * @brief Function to get the last received CMS or CME error code
+ *
+ */
+int                     at_cmd_interface_get_last_error(void);
```

Comment:
I'd much rather use the object-verb pattern: `at_cmd_interface_last_err_get()`,
if dropping the `_interface` as suggested above: `at_cmd_last_err_get()`.

Reply:
I don't agree, moving the verb to the end makes the function name strange to read.
The change will also have implications on how a vissual assit tool will be able to help you, I would rather have the tool list all objects I can get than all operations I can do on an objects, this is ofcourse personal preference. 

When I look through the nrf coude it does not seem to be any consistency, while when I look through the zephyr code it seems to maybe lean towards verb-object.